### ISSUE TITLE
error using nvidia-container-cli for enroot (bug) 

### ISSUE THREAD

**Issue by KapilS25** [NVIDIA/libnvidia-container#125]
As reported by enroot developer, kindly look into this : https://github.com/NVIDIA/enroot/issues/54#issuecomment-762169057

**Comment by klueska** [comment]
Thanks for the report. There is a long thread on that link (some of which is relevant, some of which is not). Can you summarize the exact bug you are seeing with `libnvidia-container` here.

**Comment by KapilS25** [comment]
nvidia-container-cli unable to mount /dev from host to inside the containers /dev  
https://github.com/NVIDIA/enroot/issues/54#issuecomment-762148027

**Comment by klueska** [comment]
I don't know anything about `enroot`. Do you have a simple reproducer with `nvidia-container-cli` directly that I can use to see what your issue is?

**Comment by KapilS25** [comment]
Adding @3XX0 (enroot developer) in the conversation, @3XX0 can you please explain the issue to @klueska , as i dont know how  exactly enroot start is using nvidia-container-cli.

**Comment by 3XX0** [comment]
Basically it looks like the device mount fails if the device already exists at the destination.
I've never seen this before, so this might be RHEL specific:
```
mount error: file creation failed: /scratch/pbs/enroot-data/user-613.chas052/lammps/dev/nvidia-uvm-tools: operation not permitted
```

`/dev/nvidia-uvm-tools` already exists because `/dev` is bind mounted in the container, so mount should try to create it.

@KapilS25 Can you try adding `strace` to `nvidia-container-cli` in the nvidia hook so we can see the exact failure on `open`

**Comment by KapilS25** [comment]
Please find the attached output file for nvidia-container-cli with strace.
[dev_mount_issue_nvidia-container-cli.strace.txt](https://github.com/NVIDIA/libnvidia-container/files/5834159/dev_mount_issue_nvidia-container-cli.strace.txt)

**Comment by 3XX0** [comment]
Thanks, this makes sense now, the umask will make the open fail as it tries to adjust permissions

**Comment by klueska** [comment]
So it sounds like this is not actually a bug in `libnvidia-container` then, but rather expected behaviour given the `umask` set on `/dev/nvidia-uvm-tools`.

**Comment by 3XX0** [comment]
It is a bug, the file exists and can just be mounted over.
Libnvidia-container shouldn't try to adjust the permission of a device file to reflect the system umask. Permissions of the underlying file actually don't really matter.

**Comment by ydm-amazon** [comment]
I've also encountered the same bug
